src/aap_eda/core/models/source.py

```diff
@@ -0,0 +1,50 @@
+#  Copyright 2022 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import uuid
+
+from django.db import models
+
+from aap_eda.core.enums import RestartPolicy
+
+__all__ = "Source"
+
+
+class Source(models.Model):
+    class Meta:
+        db_table = "core_source"
+        indexes = [
+            models.Index(fields=["uuid"], name="ix_source_uuid"),
+            models.Index(fields=["name"], name="ix_source_name"),
+        ]
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4)
+    type = models.TextField(null=False, default="", blank=True)
+    name = models.TextField(null=False, unique=True)
+    args = models.TextField(null=True, default=None)
```

Comment:
Why this is not a json field?

Reply:
As I was told, TextField (yaml format) has better performance in DB.

Reply:
The serializer operates in the context of a request. We can support text from the view and validate/process it in the serializer, but the type of this field is an object and it's intrinsic to the model. We are not going to treat this internally as text but as object. Doesn't make sense to perform conversions all the time. I think it should be a json field.

Reply:
@Alex-Izquierdo it was a JSON field but @benthomasson had mentioned that there were performance issues with JSON conversions and storage. And we wanted to follow the extra_var which gets stored as string.

Reply:
Can we elaborate these performance issues? @benthomasson we are not going to perform queries on the content or partial writes. Either we expect large objects. I don't see any potential issue in our usecase. On the contrary, we are converting a dictionary to yaml only to convert it back again later (the case of listener_args), that is an unnecessary use of cycles.

Reply:
At least we could add a property or methods to work with this field as a dict out of the box.

Reply:
Doesn't seem like you can attach a particular serializer (such as the YAML one) to a single field.
If it "must" be yaml, maybe override the appropriate de-/serialization methods and "manually" decode/encode to/from yaml.

Reply:
JSONB decomposes the string and indexes it.  We are fine with using the JSON type.

Reply:
updated to json field